src/main.rs

```diff
@@ -27,26 +26,21 @@ fn main() {
         println!("{}", USAGE);
         exit(0);
     }
-    match Target::parse(args) {
-        Ok(target) => {
-            if !target.should_run_command() {
-                debug!("Nothing to do.");
-                exit(target.needs_rebuild.into());
-            } else {
-                match target.run_command() {
-                    Ok(code) => exit(code),
-                    Err(e) => {
-                        eprintln!("{}", e);
-                        exit(1);
-                    }
-                }
-            }
-        }
-        Err(e) => {
-            eprintln!("{}", e);
-            exit(1);
-        }
+
+    let target = Target::parse(args).unwrap_or_else(|e| {
+        eprintln!("{}", e);
+        exit(1);
+    });
+    if !target.should_run_command() {
```

Comment:
What I'd _really_ do is have the call to `run_command` internally handle the `should_run_command` conditional. Then it could look something like

```rust
Target::parse(args)
  .and_then(|t| t.run())
  .unwrap_or_else(|e| {
    eprintln!("{e}");
    exit(1);
  });
```